test/puppetlabs/http/client/async_plaintext_test.clj

```diff
@@ -352,4 +356,110 @@
                                  {:socket-timeout-milliseconds 2000})]
               (let [response @(common/get client url {:as :text})]
                 (is (= 200 (:status response)))
-                (is (= "Hello, World!" (:body response)))))))))))
\ No newline at end of file
+                (is (= "Hello, World!" (:body response)))))))))))
+
+(defn- build-content-handler
+  [data initial-bytes-read? wait-forever?]
+  (fn [_]
+    (let [outstream (PipedOutputStream.)
+          instream (PipedInputStream. 4)]
+      (.connect instream outstream)
+      ;; Return the response immediately and asynchronously stream some data into it
+      (future
+       (.write outstream (.getBytes data))
+       (.flush outstream)
+       (if wait-forever?
+         ; The :socket-timeout-milliseconds setting on the client means we don't actually block forever and forces
+         ; a SocketTimeoutException on the underlying socket
+         (deref (promise)))
+       ; Block until the client confirms it has read the first few bytes
+       ; Again :socket-timeout-milliseconds on the client ensures we can't really get stuck here, even if the
+       ; test fails
+       (if initial-bytes-read?
+         (deref initial-bytes-read?))
+       ; Write the last of the data
+       (.write outstream (.getBytes "xxxx"))
+       (.close outstream))
+      {:status 200
+       :body instream})))
+
+(defn- clojure-streaming-success
+  [data-size opts]
+  (testing (str "clojure streaming success: " data-size " bytes with opts " opts)
+    (let [data (apply str (repeat (- data-size 4) "x"))
+          ;; If :unbuffered-stream is enabled then we check that we can read some bytes from the response before all
+          ;; bytes have actually been transmitted
+          ;; We need to make sure the amount of data sent is enough to ensure it doesn't get buffered by the OS or Jetty
+          ;; About 64K seems to be the threshold
+          initial-bytes-read? (if (and (= :unbuffered-stream (:as opts))
+                                       (not (:decompress-body opts))
```

Comment:
Yes, thanks to badly chosen super-compressible test data on my part. As it stood, the compressed payload was too small to fill the internal buffers and so never triggered actual streaming.

I reworked the tests to use data that doesn't compress quite so pathologically so that streaming-with-compression is properly validated.